field.cpp

```diff
@@ -0,0 +1,194 @@
+#include "field.h"
+#include <random>
+#include <utility>
+
+Field::Field(int fieldSize){
+    setSize(fieldSize);
+    isMovable = true;
+}
+
+void Field::setSize(int fieldSize){
+    if(size != fieldSize){
+        size = fieldSize;
+
+        field.resize(size);
+        for (int i = 0; i < size; i++) {
+            field[i].resize(size);
+        }    size = fieldSize;
+    }
+}
+
+void Field::createField(int fieldSize){
+    setSize(fieldSize);
+    initializeField();
+}
+
+void Field::updateMovability(){
+    /*
+    * Field is movable if at least one cell is movable
+    */
+    for(int i = 0; i<size; i++){
+        for(int j = 0; j < size; j++){
```

Comment:
Find the STL algorithm that will check this without manual loop through the vector

Reply:
done